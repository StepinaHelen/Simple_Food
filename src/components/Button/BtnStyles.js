import styled from "styled-components";

export const Btn = styled.button`
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
  &.login {
    padding: 9px 51px;
  }
  &:disabled {
    opacity: 0.7;
  }
  &:hover {
    background-color: #c913d6;
  }
  &.categories {
    margin-bottom: 12px;
    padding: 4px 12px;
    margin-right: 12px;
    @media screen and (min-width: 768px) {
      padding: 8px 23px;
      margin-right: 24px;
    }
  }

  &.cardItem {
    font-size: 16px;
    line-height: 19px;
    padding: 5px 27px;
  }

  &.order {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &.goToOrder {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 13px 54px;
    font-weight: 600;
  }
`;
