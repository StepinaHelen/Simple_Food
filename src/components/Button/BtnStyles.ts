import styled from "styled-components";

export const Btn = styled.button`
  background-color: ${({ theme }) => theme.colors.primary};
  padding: 5px 27px;
  ${({ theme }) => theme.fontText({})}
  color: ${({ theme }) => theme.colors.btnColor};
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
  &.login {
    padding: 9px 51px;
  }
  &:disabled {
    opacity: 0.7;
  }
  &:hover {
    background-color: ${({ theme }) => theme.colors.secondary};
  }
  &.categories {
    margin-bottom: 12px;
    padding: 4px 12px;
    margin-right: 12px;
    @media screen and ${({ theme }) => theme.media.medium} {
      padding: 8px 23px;
      margin-right: 24px;
    }
  }

  &.cardItem {
    ${({ theme }) => theme.fontText({})}
    padding: 5px 27px;
  }

  &.order {
    ${({ theme }) =>
      theme.flex({
        alignItems: "center",
        justifyContent: "center",
      })}
  }

  &.goToOrder {
    ${({ theme }) =>
      theme.flex({
        alignItems: "center",
        justifyContent: "center",
      })}
    padding: 13px 54px;
    font-weight: 600;
  }
`;
