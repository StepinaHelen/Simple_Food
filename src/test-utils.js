import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
