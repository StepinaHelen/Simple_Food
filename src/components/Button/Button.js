import { Btn } from "./BtnStyles";

const Button = (props) => {
  return (
    <Btn
      type={props.type || "button"}
      className={props.className}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.children}
    </Btn>
  );
};

export default Button;
