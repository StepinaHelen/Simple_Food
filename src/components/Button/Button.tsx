import { Btn } from "components/Button/BtnStyles";
import { IBtnProps } from "common/interfaces";

const Button = (props: IBtnProps) => {
  return (
    <Btn
      type={props.type || "button"}
      className={props.className}
      onClick={(e) => props.onClick?.(e)}
      disabled={props.disabled}
    >
      {props.children}
    </Btn>
  );
};

export default Button;
