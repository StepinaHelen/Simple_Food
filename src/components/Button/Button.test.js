import { render } from "../../test-utils";
import Button from "./Button";
describe("test Button", () => {
  it("Button renders", () => {
    render(<Button />);
  });
  it("Button snapshot", () => {
    const view = render(<Button />);
    expect(view).toMatchSnapshot();
  });
});
