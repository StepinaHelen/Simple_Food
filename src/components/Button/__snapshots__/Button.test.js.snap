// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button Button snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c0.login {
  padding: 9px 51px;
}

.c0:disabled {
  opacity: 0.7;
}

.c0:hover {
  background-color: #c913d6;
}

.c0.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c0.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c0.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

@media screen and (min-width:768px) {
  .c0.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

<body>
    <div>
      <button
        class="c0"
        type="button"
      />
    </div>
  </body>,
  "container": .c0 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c0.login {
  padding: 9px 51px;
}

.c0:disabled {
  opacity: 0.7;
}

.c0:hover {
  background-color: #c913d6;
}

.c0.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c0.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c0.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

@media screen and (min-width:768px) {
  .c0.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

<div>
    <button
      class="c0"
      type="button"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
